src/test/java/io/cdap/plugin/sfmc/sink/DataExtensionRecordWriterTest.java

```diff
@@ -0,0 +1,115 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.sfmc.sink;
+
+import com.exacttarget.fuelsdk.ETDataExtensionColumn;
+import com.exacttarget.fuelsdk.ETDataExtensionRow;
+import com.exacttarget.fuelsdk.ETResponse;
+import com.exacttarget.fuelsdk.ETResult;
+import com.exacttarget.fuelsdk.ETSdkException;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataExtensionRecordWriterTest {
+
+  @Test
+  public void testWriteWInsertWNullResponse() throws ETSdkException, IOException {
+    StructuredRecord record = Mockito.mock(StructuredRecord.class);
+    List<ETDataExtensionRow> batch = new ArrayList<>();
+    ETDataExtensionRow row1 = new ETDataExtensionRow();
+    row1.setId("121");
+    row1.setDataExtensionKey("DE");
+    ETDataExtensionRow row2 = new ETDataExtensionRow();
+    row1.setId("122");
+    batch.add(row1);
+    batch.add(row2);
+    batch.size();
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    ETDataExtensionColumn column = new ETDataExtensionColumn();
+    column.setLength(123);
+    column.setName("Name");
+    columnList.add(column);
+    DataExtensionClient dataExtensionClient = Mockito.mock(DataExtensionClient.class);
+    DataExtensionInfo info = new DataExtensionInfo("externalKey", columnList);
+    RecordDataExtensionRowConverter recordDataExtensionRowConverter = new RecordDataExtensionRowConverter(info, true);
+    DataExtensionRecordWriter dataExtensionRecordWriter = new DataExtensionRecordWriter(dataExtensionClient,
+      recordDataExtensionRowConverter, Operation.INSERT, 1, false);
+    dataExtensionRecordWriter.write(null, record);
+  }
+
+
+  @Test
+  public void testWriteWUpsert() throws ETSdkException, IOException {
+    TaskAttemptContext context = Mockito.mock(TaskAttemptContext.class);
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    ETDataExtensionColumn column = new ETDataExtensionColumn();
+    column.setLength(123);
+    column.setName("Name");
+    columnList.add(column);
+    DataExtensionClient dataExtensionClient = Mockito.mock(DataExtensionClient.class);
+    DataExtensionInfo info = new DataExtensionInfo("externalKey", columnList);
+    RecordDataExtensionRowConverter recordDataExtensionRowConverter = new RecordDataExtensionRowConverter(info, true);
+    DataExtensionRecordWriter dataExtensionRecordWriter = new DataExtensionRecordWriter(dataExtensionClient,
+      recordDataExtensionRowConverter, Operation.UPSERT, 1, false);
+    dataExtensionRecordWriter.close(context);
```

Comment:
Looks like you're just creating a DataExtensionRecordWriter object and closing it, not really testing anything.
Same comment for other tests in this class

Reply:
Added the validation.